### ISSUE TITLE
[rocm-opencl-runtime] clinfo and libamdocl64.so  contain references to $srcdir

### ISSUE THREAD

**Issue by tpkessler** [rocm-arch/rocm-arch#95]
`clinfo` links `libOpenCL` from $srcdir. In my case:

```
$ ldd /opt/rocm/bin/x86_64/clinfo
	linux-vdso.so.1 (0x00007ffc01bf8000)
	libOpenCL.so.1 => /home/torsten/Dokumente/rocm-arch/rocm-opencl-runtime/src/rocm-opencl-runtime/build/api/opencl/khronos/icd/libOpenCL.so.1 (0x00007f74dd829000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f74dd5f8000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f74dd5de000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f74dd418000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f74dd3f6000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f74dd3ef000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f74dd2a9000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f74dd848000)
```

**Comment by rigred** [comment]
The issue is once again the same as #57 
https://github.com/rocm-arch/rocm-arch/commit/2b0595f3be438a773bcd41e9e603b4b65d535e85

**Comment by rigred** [comment]
As far as I can tell this is resolved?
![image](https://user-images.githubusercontent.com/5716332/79252099-5fb80d80-7e81-11ea-8bef-fdac56c211ac.png)

**Comment by tpkessler** [comment]
Not for me:

```
$ ldd /opt/rocm/bin/x86_64/clinfo
	linux-vdso.so.1 (0x00007ffe6d9cb000)
	libOpenCL.so.1 => /home/torsten/Dokumente/rocm-arch/rocm-opencl-runtime/src/rocm-opencl-runtime/build/api/opencl/khronos/icd/libOpenCL.so.1 (0x00007fde3c40c000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fde3c1db000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fde3c1c1000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007fde3bffb000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007fde3bfd9000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x00007fde3bfd2000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007fde3be8c000)
	/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fde3c429000)
```

**Comment by tpkessler** [comment]
According to the the [cmake wiki](https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/RPATH-handling)

>By default if you don't change any RPATH related settings, CMake will link the executables and shared libraries with full RPATH to all used libraries in the build tree. When installing, it will clear the RPATH of these targets so they are installed with an empty RPATH.

So `clinfo` should not depend on `$srcdir`. I think this is an upstream issue.

**Comment by ranisalt** [comment]
> So `clinfo` should not depend on `$srcdir`. I think this is an upstream issue.

`clinfo` shouldn't even exist in this package. It can be removed, just use `clinfo` from [community]

**Comment by rigred** [comment]
Just checked and confirmed that indeed rocm-opencl-runtime includes clinfo
![image](https://user-images.githubusercontent.com/5716332/80013098-a1bf0000-84ce-11ea-8327-3c774d57ae22.png)

The rocm clinfo also outputs quite different data vs the system clinfo
I have attached both outputs here for comparison:
[rocm_clinfo.txt](https://github.com/rocm-arch/rocm-arch/files/4517923/rocm_clinfo.txt)
[sys_clinfo.txt](https://github.com/rocm-arch/rocm-arch/files/4517925/sys_clinfo.txt)

**Comment by ranisalt** [comment]
> The rocm clinfo also outputs quite different data vs the system clinfo

In my system, while the output format is different, the data is the same, and quickly glancing at your files it seems to be the case for you too. It displays roughly the same information, but in a different order and with a different formatting.

I don't get why AMD rolled their own clinfo, with different formatting to the same data.

**Comment by tpkessler** [comment]
Resolved with 3.5.0.
